From 4a3fb506d7b66c93191aeedc211446ccb44c9572 Mon Sep 17 00:00:00 2001
From: Michael Ellerman <mpe@ellerman.id.au>
Date: Thu, 11 Jan 2018 16:33:59 +1100
Subject: [PATCH 2/6] powerpc/pseries/rfi-flush: Drop PVR-based selection
Patch-mainline: Submitted, 2018-01-09 LKML + linuxppc-dev
References: bsc#1075088

This does not work as intended. On non-updated firmware these
instructions are nops, and do nothing. If the firmware is not
advertising the support then we should assume it does not have these
instructions enabled, and use the fallback flush.

Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
Reviewed-by: Torsten Duwe <duwe@suse.de>
---
 arch/powerpc/platforms/pseries/setup.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/arch/powerpc/platforms/pseries/setup.c b/arch/powerpc/platforms/pseries/setup.c
index 0950df25ae0b..7dc450740867 100644
--- a/arch/powerpc/platforms/pseries/setup.c
+++ b/arch/powerpc/platforms/pseries/setup.c
@@ -397,15 +397,8 @@ static void pSeries_setup_rfi_flush(void)
 		if (!(behaviour & H_GET_CPU_CHAR_BEHAV_L1_FLUSH_LOW_PRIV))
 			enable = false;
 	} else {
-		if (__is_processor(PVR_POWER7) || __is_processor(PVR_POWER7p))
-			types = L1D_FLUSH_NONE;
-		else if (__is_processor(PVR_POWER8E) || __is_processor(PVR_POWER8NVL) ||
-				__is_processor(PVR_POWER8))
-			types = L1D_FLUSH_ORI;
-		else {
-			/* Default to fallback if case hcall is not available */
-			types = L1D_FLUSH_FALLBACK;
-		}
+		/* Default to fallback if case hcall is not available */
+		types = L1D_FLUSH_FALLBACK;
 	}
 
 	setup_rfi_flush(types, enable);
-- 
2.14.3

